src/java.base/share/classes/java/lang/doc-files/ValueBased.html

```diff
@@ -31,36 +31,39 @@
-<code>java.time.LocalDateTime</code>, are <em>value-based</em>.  Instances of a
-value-based class:
+Some classes, such as <code>java.lang.Integer</code> and
+<code>java.time.LocalDate</code>, are <em>value-based</em>.
+A value-based class has the following properties:
 <ul>
-    <li>are final and immutable (though may contain references to mutable
-        objects);</li>
-    <li>have implementations of <code>equals</code>,
+    <li>the class declares only final instance fields (though these may contain references
+        to mutable objects);</li>
+    <li>the class declares implementations of <code>equals</code>,
         <code>hashCode</code>, and <code>toString</code> which are computed
-        solely from the instance's state and not from its identity or the state
-        of any other object or variable;</li>
-    <li>make no use of identity-sensitive operations such as reference
-        equality (<code>==</code>) between instances, identity hash code of
-        instances, or synchronization on an instances's intrinsic lock;</li>
-    <li>are considered equal solely based on <code>equals()</code>, not
-        based on reference equality (<code>==</code>);</li>
-    <li>do not have accessible constructors, but are instead instantiated
-        through factory methods which make no commitment as to the identity
-        of returned instances;</li>
-    <li>are <em>freely substitutable</em> when equal, meaning that interchanging
-        any two instances <code>x</code> and <code>y</code> that are equal
-        according to <code>equals()</code> in any computation or method
-        invocation should produce no visible change in behavior.
-    </li>
+        solely from the values of the class's instance fields (and properties of
+        the objects they reference), not from the instance's identity;</li>
+    <li>the class's methods treat instances as <em>freely substitutable</em>
+        when equal, meaning that interchanging any two instances <code>x</code> and
+        <code>y</code> that are equal according to <code>equals()</code> produces no
+        visible change in the behavior of the class's methods;</li>
+    <li>the class performs no synchronization using an instance's monitor;</li>
+    <li>the class does not declare (or has deprecated any) accessible constructors;</li>
+    <li>the class does not provide any other instance creation mechanism that promises
+        a unique identity on each method call&mdash;in particular, any factory
+        method must allow for the possibility that if two independently-produced
+        instances are equal according to <code>equals()</code>, they may also be
+        equal according to <code>==</code>;</li>
+    <li>the class is final, and extends either <code>Object</code> or a hierarchy of
+        abstract classes that declare no instance fields or instance initializers
+        and whose constructors are empty.</li>
+    <li>
 </ul>
 
```

Comment:
Perhaps as an intro to the following points to make it clear these are about what a program should and should not do.
Should use equals(); should use explicit synchronization using lock objects or instances that are not value-based classes, etc.
`Programs should not attempt to distinguish the identities of value based class instances, otherwise the result may be unpredictable.`

Reply:
I'm not totally following. This is a revision to the "When two instances" paragraph? Can you propose an alternative phrasing for the entire paragraph?

Reply:
These two paragraphs are fine, they express the negative, not what a developer should do.